windows/devices/uwb/UwbCxAdapter.cxx

```diff
@@ -0,0 +1,239 @@
+        { ::uwb::protocol::fira::UwbConfiguration::ParameterTag::ControllerMacAddress, UWB_APP_CONFIG_PARAM_TYPE_DEVICE_MAC_ADDRESS }, // TODO this field only applies when the host is the controller
+        { ::uwb::protocol::fira::UwbConfiguration::ParameterTag::SlotsPerRr, UWB_APP_CONFIG_PARAM_TYPE_SLOTS_PER_RR },
+        { ::uwb::protocol::fira::UwbConfiguration::ParameterTag::MaxContentionPhaseLength, UWB_APP_CONFIG_PARAM_TYPE_SLOTS_PER_RR }, // TODO figure out this field
+        { ::uwb::protocol::fira::UwbConfiguration::ParameterTag::SlotDuration, UWB_APP_CONFIG_PARAM_TYPE_SLOT_DURATION },
+        { ::uwb::protocol::fira::UwbConfiguration::ParameterTag::RangingInterval, UWB_APP_CONFIG_PARAM_TYPE_RANGING_INTERVAL },
+        { ::uwb::protocol::fira::UwbConfiguration::ParameterTag::KeyRotationRate, UWB_APP_CONFIG_PARAM_TYPE_KEY_ROTATION_RATE },
+        { ::uwb::protocol::fira::UwbConfiguration::ParameterTag::MacFcsType, UWB_APP_CONFIG_PARAM_TYPE_MAC_FCS_TYPE },
+        { ::uwb::protocol::fira::UwbConfiguration::ParameterTag::MaxRrRetry, UWB_APP_CONFIG_PARAM_TYPE_MAX_RR_RETRY },
+    };
+} // namespace windows::devices::uwb::detail
+
+using namespace windows::devices::uwb::detail;
+using namespace windows::devices;
+
+template <class... Ts>
+struct overloaded : Ts...
+{
+    using Ts::operator()...;
+};
+// explicit deduction guide (not needed as of C++20)
+template <class... Ts>
+overloaded(Ts...) -> overloaded<Ts...>;
+
+std::unique_ptr<uint8_t[]>
+windows::devices::uwb::GenerateUwbSetAppConfigParameterDdi(const ::uwb::protocol::fira::UwbSessionData& uwbSessionData)
+{
+    UwbSetAppConfigurationParametersBuilder builder{ uwbSessionData.sessionId };
+
+    // TODO NUMBER_OF_CONTROLEES
+    // TODO DEVICE_MAC_ADDRESS
+    // TODO DST_MAC_ADDRESS
+    // TODO DEVICE_TYPE
+    const auto sessionUwbMap = uwbSessionData.uwbConfiguration.GetValueMap();
+
+    for (const auto [ptag, value] : sessionUwbMap) {
+        std::visit(overloaded{
+                       [&builder, ptag](bool arg) {
+                           builder.Push(std::move(UwbAppConfigurationParameter{ uint8_t(arg), service_ddi_map.at(ptag) }));
+                       },
+                       [&builder, ptag](uint8_t arg) {
+                           builder.Push(std::move(UwbAppConfigurationParameter{ uint8_t(arg), service_ddi_map.at(ptag) }));
+                       },
+                       [&builder, ptag](uint16_t arg) {
+                           builder.Push(std::move(UwbAppConfigurationParameter{ uint16_t(arg), service_ddi_map.at(ptag) }));
+                       },
+                       [&builder, ptag](uint32_t arg) {
+                           builder.Push(std::move(UwbAppConfigurationParameter{ uint32_t(arg), service_ddi_map.at(ptag) }));
+                       },
+                       [&builder, ptag](::uwb::protocol::fira::Channel arg) {
+                           uint8_t data = 9;
```

Comment:
If this is supposed to be the default value, then it should probably be assigned in a `default` case label. However, if nothing matches, I'd think we'd want to flag that somehow, not assign a default value.

Reply:
Currently the switch statement does cover all the enum cases. Maybe I'll just throw an error if nothing matches (shouldn't happen)

Reply:
This becomes moot if we switch to use `notstd::to_underlying`.